### PR TITLE
Pick Search provider from search app

### PR DESCRIPTION
**Pull request by daleharvey**

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by try-server-hook** [comment]
[daleharvey Dale  Harvey (daleharvey)](https://github.com/daleharvey) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=843c96d7d7b1)

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by try-server-hook** [comment]
[daleharvey Dale  Harvey (daleharvey)](https://github.com/daleharvey) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=06e2ee97ea38)

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by try-server-hook** [comment]
[daleharvey Dale  Harvey (daleharvey)](https://github.com/daleharvey) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=ce95f9442314)

File: apps/search/index.html [hunk 1]
```diff
@@ -93,10 +93,15 @@ <h2 id="places-header" class="hidden"><span data-l10n-id="history"></span></h2>
                aria-label="Places"></section>
 
       <div id="suggestions-wrapper">
-        <h2><span id="suggestion-provider"></span></h2>
+        <h2>
+          <select id="suggestions-select"></select>
+          <label for="suggestions-select" id="suggestions-provider"></label>
+          <div class="link-triangle"> </div>
```
Comment (KevinGrandon) [inline apps/search/index.html]: Is this used anywhere? Can we remove it?

Comment (daleharvey) [inline apps/search/index.html]: Its that little triangle on the bottom right to make it look a bit more like a select box

File: apps/search/style/search.css [hunk 2]
```diff
@@ -78,11 +97,19 @@ h2 {
...
 }
 
-h2 span {
+/* The select box is hidden behind search provider label since we
+    only want to see the dialog prompt, not the actual select box  */
+#suggestions-select {
+  position: absolute;
+  top: -0.9rem;
+  height: 1.7rem;
+}
```
Comment (benfrancis) [inline apps/search/style/search.css]: I can still see the select box peeking out from behind the title

File: apps/search/js/providers/suggestions.js [hunk 3]
```diff
@@ -5,11 +5,24 @@
...
+
+    var selected = SearchProvider.selected();
+    var providers = SearchProvider.providers();
+    var selectHTML = Object.keys(providers).map(provider => {
+      return '<option value="' + provider + '" ' +
+        ((provider === selected) ? 'selected="selected"' : '') +
+        '>' + providers[provider].title + '</option>';
+    }).join('');
+
+    suggestionsSelect.innerHTML = selectHTML;
```
Comment (benfrancis) [inline apps/search/js/providers/suggestions.js]: nit: Might there be a way of doing this without using innerHTML?

File: apps/search/index.html (hunk 1 above)
Comment (KevinGrandon) [inline apps/search/index.html]: Oh, I assumed that the ```::before``` selector was handling this. Where is the css/js that references ```.link-triangle```?

Comment (daleharvey) [inline apps/search/index.html]: You were right, was using an extra div before then moved to .link-triangle, removed it

**Pull request closed without merging**